test/Destructurama.Attributed.Tests/NotLoggedIfDefaultAttributeTests.cs

```diff
@@ -0,0 +1,212 @@
+﻿using Destructurama.Attributed.Tests.Support;
+using NUnit.Framework;
+using Serilog;
+using Serilog.Events;
+using System;
+using System.Linq;
+
+namespace Destructurama.Attributed.Tests
+{
+    struct NotLoggedIfDefaultStruct
+    {
+        public int Integer { get; set; }
+
+        public DateTime DateTime { get; set; }
+    }
+
+    struct NotLoggedIfDefaultStructWithAttributes
+    {
+        [NotLoggedIfDefault]
+        public int Integer { get; set; }
+
+        [NotLoggedIfDefault]
+        public DateTime DateTime { get; set; }
+
+        public int IntegerLogged { get; set; }
+
+        public DateTime DateTimeLogged { get; set; }
+    }
+
+    class NotLoggedIfDefaultCustomizedDefaultLogs
+    {
+        [NotLoggedIfDefault]
+        public string String { get; set; }
+
+        [NotLoggedIfDefault]
+        public int Integer { get; set; }
+
+        [NotLoggedIfDefault]
+        public int? NullableInteger { get; set; }
+
+        [NotLoggedIfDefault]
+        public object Object { get; set; }
+
+        [NotLoggedIfDefault]
+        public object IntegerAsObject { get; set; }
+
+        [NotLoggedIfDefault]
+        public DateTime DateTime { get; set; }
+
+        [NotLoggedIfDefault]
+        public NotLoggedIfDefaultStruct Struct { get; set; }
```

Comment:
I mean `public object Struct { get; set; }` and then `Struct = new NotLoggedIfDefaultStruct(...)`

Reply:
With my recent changes (`cc0fd2b`) the type that is evaluated is the Property's defined type, not the actual one (in case of boxing).

In orded to reduce the cost I can introduce a property on the attribute, eg. `Unbox = true`, that would check for the actual (unboxed) type of `value`.

What do you think? 🙂

Reply:
probably not worth it